### PR TITLE
Convert lib/reversetunnel to use slog

### PR DESCRIPTION
**Pull request by rosstimothy**

### CODE REVIEW

File: lib/reversetunnel/agent.go [hunk 1]
```diff
@@ -555,10 +560,10 @@ func (a *agent) handleDrainChannels() error {
 			bytes, _ := a.clock.Now().UTC().MarshalText()
 			_, err := a.hbChannel.SendRequest(a.ctx, "ping", false, bytes)
 			if err != nil {
-				a.log.Error(err)
+				a.logger.ErrorContext(a.ctx, "failed to sing ping request", "error", err)
```
Comment (marcoandredinis) [inline lib/reversetunnel/agent.go:563]: 🧑‍🎤 🎤

File: lib/reversetunnel/remotesite.go [hunk 2]
```diff
@@ -668,17 +669,16 @@ func (s *remoteSite) watchCertAuthorities(remoteWatcher *services.CertAuthorityW
 		return trace.Wrap(err)
 	}
 
-	s.logger.Debugf("Watching for cert authority changes.")
+	s.logger.DebugContext(s.ctx, "Watching for cert authority changes")
 	for {
 		select {
 		case <-s.ctx.Done():
-			s.logger.WithError(s.ctx.Err()).Debug("Context is closing.")
```
Comment (marcoandredinis) [inline lib/reversetunnel/remotesite.go:675]: Can we check the `s.ctx.Err() != nil` log if true?

**Review by marcoandredinis — APPROVED** [review]

File: lib/reversetunnel/localsite.go [hunk 3]
```diff
@@ -788,9 +802,9 @@ func (s *localSite) handleHeartbeat(rconn *remoteConn, ch ssh.Channel, reqC <-ch
 
 			log := logger
 			if roundtrip != 0 {
-				log = logger.WithField("latency", roundtrip.String())
+				log = logger.With("latency", logutils.StringerAttr(roundtrip))
 			}
-			log.Debugf("Ping <- %v", rconn.conn.RemoteAddr())
+			log.DebugContext(ctx, "Received ping request", "remote_addr", logutils.StringerAttr(rconn.conn.RemoteAddr()))
```
Comment (espadolini) [inline lib/reversetunnel/localsite.go:807]: Is there a better way to do conditional fields in slog?

File: lib/reversetunnel/localsite.go [hunk 4]
```diff
@@ -788,9 +802,9 @@ func (s *localSite) handleHeartbeat(rconn *remoteConn, ch ssh.Channel, reqC <-ch
 
 			log := logger
 			if roundtrip != 0 {
-				log = logger.WithField("latency", roundtrip.String())
+				log = logger.With("latency", logutils.StringerAttr(roundtrip))
```
Comment (espadolini) [inline lib/reversetunnel/localsite.go:805]: Sould we add special handling for `time.Duration` to our handlers so we don't have to do this every time, kinda like we do for `error`?

**Review by espadolini — APPROVED** [review]

**Pull request merged**